v2/util/gen-templates.js

```diff
@@ -45,10 +52,50 @@ async function genSlideTemplate(lang) {
 
     $Template('.swiper-wrapper').append(slide);
     $Template('.swiper-wrapper').append(index === length - 1 ? '\n      ' : '\n        ');
+
+    // images will be downloaded and compressed only when they do not exist yet
+    if (!picsExists.includes(pic.split('/header-images/')[1])) {
+      pics.add(pic);
+    }
+  });
+
+  writeFile(`./templates/posts-slide-${lang}.tt2`, $Template.html() + '\n');
+};
+
+Promise.all([
+  genSlideTemplate('en'),
+  genSlideTemplate('cn'),
+]).then(() => {
+  for(pic of pics) {
+    (async() => {
```

Comment:
This loop will create several Async Functions, run independently with each other. That means these lines has no real meaning.

Reply:
changed.